sdk/rust/nym-sdk/examples/tcp_proxy.rs

```diff
@@ -0,0 +1,96 @@
+use nym_sdk::tcp_proxy;
+use rand::Rng;
+use tokio::io::AsyncWriteExt;
+use tokio::net::{TcpListener, TcpStream};
+use tokio::task;
+use tokio_stream::StreamExt;
+use tokio_util::codec;
+
+// This is a basic example which opens a single TCP connection and streams between a client and
+// server, so only uses a single session under the hood. See <MORE ELABORATE EXAMPLE> for use of
+// multiple sessions/streams.
+#[tokio::main]
+async fn main() {
+    // Comment this out to just see println! statements from this example.
+    // Nym client logging is very informative but quite verbose.
+    //
+    // Run with RUST_LOG="debug" to see the Message Decay related logging if you want
+    // a better idea of the internals of the proxy message ordering.
+    nym_bin_common::logging::setup_logging();
+
+    let upstream_tcp_addr = "127.0.0.1:9067";
+    let mut proxy_server = tcp_proxy::NymProxyServer::new(upstream_tcp_addr, "~/tmp/server_client")
+        .await
+        .unwrap();
+    let proxy_nym_addr = proxy_server.nym_address();
+    let proxy_client = tcp_proxy::NymProxyClient::new(*proxy_nym_addr, "127.0.0.1", "8080", 60)
+        .await
+        .unwrap();
+
+    task::spawn(async move {
+        let _ = proxy_server.run_with_shutdown().await;
+    });
+
+    task::spawn(async move {
+        let _ = proxy_client.run().await;
+    });
+
+    // 'Server side' thread: just echoing back received messages.
+    task::spawn(async move {
```

Comment:
We can use `tokio` spawn directly:

```suggestion
    tokio::spawn(async move {
```